processors/hra/extractors.py

```diff
@@ -19,79 +19,43 @@ def extract_source(record):
     return source
 
 
-def extract_trial(record):
+def extract_publications(record):
 
-    # Get identifiers
-    identifiers = base.helpers.clean_dict({
-        'nct': _clean_identifier(record['nct_id'], prefix='NCT'),
-        'isrctn': _clean_identifier(record['isrctn_id'], prefix='ISRCTN'),
-        'hra': record['hra_id'],
-    })
-
-    # Get public title
-    public_title = base.helpers.get_optimal_title(
+    # Get title
+    title = base.helpers.get_optimal_title(
         record['application_title'],
         record['application_full_title'],
         record['hra_id'])
 
-    # Compose trial
-    trial = {
-        'primary_register': 'Health Research Authority',
-        'primary_id': record['hra_id'],
-        'identifiers': identifiers,
-        'registration_date': record['publication_date'],
-        'public_title': public_title,
-        'brief_summary': record['research_summary'],
-        'scientific_title': record['application_full_title'],
-        'study_type': record['study_type'],
-    }
-
-    return trial
-
-
-def extract_conditions(record):
-    conditions = []
-    return conditions
-
-
-def extract_interventions(record):
-    interventions = []
-    return interventions
-
-
-def extract_locations(record):
-    locations = []
-    return locations
-
-
-def extract_organisations(record):
-    organisations = []
-    organisations.append({
-        'name': record['sponsor_org'],
-        'role': 'primary_sponsor',
+    # Get abstract
+    abstract = record['research_summary'] or title
+    identifiers = base.helpers.clean_list({
+        _clean_identifier(record['nct_id'], prefix='NCT'),
+        _clean_identifier(record['euctr_id'], prefix='EUCTR'),
+        _clean_identifier(record['isrctn_id'], prefix='ISRCTN'),
     })
-    organisations.append({
-        'name': record['establishment_org'],
+    if identifiers:
+        abstract = '%s [%s]' % (abstract, '/'.join(identifiers))
```

Comment:
Good idea to add the identifiers without changing the publications' schema :+1:

In the future, we should probably add those identifiers to the related trial itself.